cmd/argocd/commands/login.go

```diff
@@ -48,7 +48,9 @@ func NewLoginCommand(globalClientOpts *argocdclient.ClientOptions) *cobra.Comman
 				}
 			}
 
-			username, password = cli.PromptCredentials()
+			if username == "" || password == "" {
+				username, password = cli.PromptCredentials()
+			}
```

Comment:
Lets change prompt credentials so that one or the other could be supplied.
```
// PromptCredentials is a helper to prompt the user for a username and password
func PromptCredentials(username, password string) (string, string) {
	for username == "" {
		reader := bufio.NewReader(os.Stdin)
		fmt.Print("Username: ")
		usernameRaw, err := reader.ReadString('\n')
		errors.CheckError(err)
		username = strings.TrimSpace(usernameRaw)
	}
	for password == "" {
		fmt.Print("Password: ")
		passwordRaw, err := terminal.ReadPassword(syscall.Stdin)
		errors.CheckError(err)
		password = string(passwordRaw)
		if password == "" {
			fmt.Print("\n")
		}
	}
	fmt.Print("\n")
	return username, password
}
```

Reply:
@jessesuen @alexmt Worth prompting for a password confirmation, too?

Reply:
This for login and adding repository credentials, not for account creation. Account creation we do prompt for password and confirm password.

Reply:
@jessesuen mea culpa.  Is there enough overlap that maybe a set of universal `PromptForUsername(username string)` and `PromptForPassword(password string, confirm bool)` methods might apply?

Reply:
I'm fine refactoring it to promote code reuse but I don't think it needs to go in as part of this change. I would just like the CLI not to prompt if --username and/or --password is supplied from args.

Reply:
@jessesuen good suggestion! implemented